Solutions/Marain.Tenancy.Cli/Marain/Tenancy/Cli/Commands/List.cs

```diff
@@ -0,0 +1,125 @@
+﻿// <copyright file="List.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Tenancy.Cli.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using ConsoleTables;
+    using Corvus.Tenancy;
+    using McMaster.Extensions.CommandLineUtils;
+
+    /// <summary>
+    /// Lists children of the specified tenant.
+    /// </summary>
+    [Command(Name = "list", Description = "List tenants.")]
+    public class List
+    {
+        private readonly ITenantProvider tenantProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="List"/> class.
+        /// </summary>
+        /// <param name="tenantProvider">The tenant provider that will be used to retrieve the information.</param>
+        public List(ITenantProvider tenantProvider)
+        {
+            this.tenantProvider = tenantProvider;
+        }
+
+        /// <summary>
+        /// Gets or sets the tenant whose children should be retrieved.
+        /// </summary>
+        [Option(CommandOptionType.SingleOrNoValue, ShortName = "t", LongName = "tenant", Description = "The Id of the tenant to retrieve children for. Leave blank to retrieve children of the root tenant.")]
+        public string TenantId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not details should be loaded for the tenants.
+        /// </summary>
+        [Option(CommandOptionType.MultipleValue, ShortName = "p", LongName = "property", Description = "The names of properties to include.")]
```

Comment:
I did not correctly guess what would happen if I don't provide this. (I presumed it'd just show everything, but apparently it shows a list of IDs?)

Is it possible to show everything? Would that be useful?

Reply:
Not easily, because some of the properties are potentially big blobs of JSON. Potentially with a bit more thought, but this is beyond what I was trying to achieve in this iteration.